kem/hybrid/hybrid.go

```diff
@@ -0,0 +1,292 @@
+// Package hybrid defines several hybrid classical/quantum KEMs.
+//
+// KEMs are combined by simple concatenation of shared secrets, cipher texts,
+// public keys, etc, see
+//
+//   https://datatracker.ietf.org/doc/draft-ietf-tls-hybrid-design/
+//   https://nvlpubs.nist.gov/nistpubs/SpecialPublications/NIST.SP.800-56Cr2.pdf
+//
+// Note that this is only fine if the shared secret is used in its entirety
+// in a next step, such as being hashed or used as key.
+//
+// For deriving a KEM keypair deterministically and encapsulating
+// deterministically, we expand a single seed to both using SHAKE256,
+// so that a non-uniform seed (such as a shared secret generated by a hybrid
+// KEM where one of the KEMs is weak) doesn't impact just one of the KEMs.
+package hybrid
+
+import (
+	"errors"
+
+	"github.com/cloudflare/circl/kem"
+
+	"github.com/cloudflare/circl/kem/kyber/kyber512"
+	"github.com/cloudflare/circl/kem/sike/sikep434"
+
+	"github.com/cloudflare/circl/internal/sha3"
+)
+
+var ErrUninitialized = errors.New("public or private key not initialized")
+
+// Temporary Hybrid scheme for testing purposes.
+func KyberSike() kem.Scheme {
+	return kyberSike
+}
+
+var kyberSike kem.Scheme = &scheme{
+	"Temporary-Kyber-Sike",
```

Comment:
I assume temporary because `x25519` still needs to be specified as a KEM?

Reply:
Yes.